s2tbx-cache/src/main/java/org/esa/s2tbx/dataio/cache/S2CacheSizeChecking.java

```diff
@@ -1,81 +1,83 @@
 package org.esa.s2tbx.dataio.cache;
 
 import java.io.File;
+import java.util.concurrent.Executors;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
-import org.esa.snap.core.util.ThreadExecutor;
-import org.esa.snap.core.util.ThreadRunnable;
-
 public class S2CacheSizeChecking {
 
     public static final S2CacheSizeChecking INSTANCE = new S2CacheSizeChecking();
     private static final Logger logger = Logger.getLogger(S2CacheSizeChecking.class.getName());
     private boolean checkingEnable;
     private double limitSizeCache;
-    private ThreadExecutor executor;
+    private final ScheduledExecutorService executor;
+    private static final double BYTE_TO_GIGA_BYTE = 0.000000001;
 
-    private S2CacheSizeChecking(){
-        //nothing to init
+    private S2CacheSizeChecking() {
+        executor = Executors.newSingleThreadScheduledExecutor();
     }
-    
+
+
     public static S2CacheSizeChecking getInstance() {
         return INSTANCE;
     }
 
     /**
      * set the parameters of the cache size checking
-     *
      */
-    public synchronized void setParameters(boolean checkingEnable,double limitSizeCache ) {
+    public synchronized void setParameters(boolean checkingEnable, double limitSizeCache) {
         this.checkingEnable = checkingEnable;
-        this.limitSizeCache = limitSizeCache;
+        this.limitSizeCache = limitSizeCache * BYTE_TO_GIGA_BYTE;
```

Comment:
If the limit is in GB, this should instead be a GB_TO_BYTES.

Reply:
Yes and I have to update this value.